File: Hash Table/3083.Existence of a Substring in a String and Its Reverse/Solution.java
class Solution {
    public boolean isSubstringPresent(String s) {
        Set<Integer> cs = new HashSet<>();
        int n = s.length();
        for (int i = 1; i < n; i++) {
            int v = (s.charAt(i - 1) - 'a') * 26 + s.charAt(i) - 'a';
            cs.add(v);
        }
        for (int i = n - 2; i > -1; i--) {
            int v = (s.charAt(i + 1) - 'a') * 26 + s.charAt(i) - 'a';
            if (cs.contains(v)) {
                return true;
            }
        }
        return false;
    }
}

File: Hash Table/3083.Existence of a Substring in a String and Its Reverse/Solution.py
class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        n = len(s)
        cs = set()
        for i in range(1, n):
            v = (ord(s[i - 1]) - ord('a')) * 26 + ord(s[i]) - ord('a')
            cs.add(v)
        for i in range(n - 2, -1, -1):
            v = (ord(s[i + 1]) - ord('a')) * 26 + ord(s[i]) - ord('a')
            if v in cs:
                return True
        return False